AudioManagerUtilities/src/CAmSocketHandler.cpp

```diff
@@ -559,23 +569,31 @@ am_Error_e CAmSocketHandler::addTimer(const timespec & timeouts, std::function<v
         return err;
     }
 
-    static auto actionPoll = [](const pollfd pollfd, const sh_pollHandle_t handle, void* userData)
+    auto actionPoll = [this](const pollfd pollfd, const sh_pollHandle_t handle, void* userData)
     {
-        uint64_t mExpirations;
-        if (read(pollfd.fd, &mExpirations, sizeof(uint64_t)) == -1)
-        {
-            //error received...try again
-            read(pollfd.fd, &mExpirations, sizeof(uint64_t));
-        }
+        uint64_t expCnt;
+        ssize_t bytes = read(pollfd.fd, &expCnt, sizeof(expCnt));
+        if (bytes == sizeof(expCnt))
```

Comment:
Yes, looks better than before.  Retries should be infinite, not only once. 
But otherwise same comments as before.  (I can see the difficulty - you need to make the retries not block the thread, hence the polling, so a simple loop is not adequate)

Reply:
Even for the timerfd the manpage states that the data can be read only in one sequence.

Reply:
OK, the answer to my question is confirmed.